editoast/src/views/study.rs

```diff
@@ -307,15 +340,15 @@ async fn patch(
     data: Json<StudyPatchForm>,
     path: Path<(i64, i64)>,
     db_pool: Data<DbPool>,
-) -> Result<Json<StudyWithScenarios>> {
+) -> Result<Json<StudyResponse>> {
     let (project_id, study_id) = path.into_inner();
-
     let mut conn = db_pool.get().await?;
+    let mut project = Default::default();
     let study_scenarios = conn
         .transaction::<_, InternalError, _>(|conn| {
             async {
                 // Check if project exists
-                let project = match Project::retrieve_conn(conn, project_id).await? {
+                project = match Project::retrieve_conn(conn, project_id).await? {
                     None => return Err(ProjectError::NotFound { project_id }.into()),
                     Some(project) => project,
                 };
```

Comment:
Likewise, get rid of Default::default() and initialize the project in the transaction with the `let Some(project)` matching assignment.